src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,75 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  FC,
+  FormEvent,
+  useState,
+} from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+import { Movie } from '../../types/Movie';
+
+interface Props {
+  newMovieHandler: (movie: Movie) => void;
+}
+
+export const FindMovie: FC<Props> = ({ newMovieHandler }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const addNewMovie = (data: MovieData | ResponseError) => {
+    if ('Error' in data) {
+      throw new Error(data.Error);
+    }
+
+    setMovie({
+      title: data.Title,
+      description: data.Plot,
+      imdbId: data.imdbID,
+      imgUrl: data.Poster,
+      imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+    });
+  };
+
+  const handleOnSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsLoading(true);
+    getMovie(query)
+      .then(newMovie => {
+        addNewMovie(newMovie);
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsError(false);
+  };
+
+  const handleOnClick = () => {
+    if (movie) {
```

Comment:
you already handle it
![image](https://github.com/mate-academy/react_movies-list-fetch-movies/assets/91411692/750112e8-fc1c-4b7f-b6f2-ca15fb819f7b)